packages/tabs/src/views/BottomTabBar.tsx

```diff
@@ -121,24 +122,34 @@ class TabBarBottom extends React.Component<BottomTabBarProps, State> {
     visible: new Animated.Value(1),
   };
 
+  keyboardDidShowSubscription: EmitterSubscription | null = null;
+  keyboardDidHideSubscription: EmitterSubscription | null = null;
+
   componentDidMount() {
     if (Platform.OS === 'ios') {
-      Keyboard.addListener('keyboardWillShow', this._handleKeyboardShow);
-      Keyboard.addListener('keyboardWillHide', this._handleKeyboardHide);
+      this.keyboardDidShowSubscription = Keyboard.addListener(
+        'keyboardWillShow',
+        this._handleKeyboardShow
+      );
+      this.keyboardDidHideSubscription = Keyboard.addListener(
+        'keyboardWillHide',
+        this._handleKeyboardHide
+      );
     } else {
-      Keyboard.addListener('keyboardDidShow', this._handleKeyboardShow);
-      Keyboard.addListener('keyboardDidHide', this._handleKeyboardHide);
+      this.keyboardDidShowSubscription = Keyboard.addListener(
+        'keyboardDidShow',
+        this._handleKeyboardShow
+      );
+      this.keyboardDidHideSubscription = Keyboard.addListener(
+        'keyboardDidHide',
+        this._handleKeyboardHide
+      );
     }
   }
 
   componentWillUnmount() {
-    if (Platform.OS === 'ios') {
-      Keyboard.removeListener('keyboardWillShow', this._handleKeyboardShow);
-      Keyboard.removeListener('keyboardWillHide', this._handleKeyboardHide);
-    } else {
-      Keyboard.removeListener('keyboardDidShow', this._handleKeyboardShow);
-      Keyboard.removeListener('keyboardDidHide', this._handleKeyboardHide);
-    }
+    this.keyboardDidShowSubscription?.remove();
+    this.keyboardDidHideSubscription?.remove();
```

Comment:
Maybe you want to keep backwards compatibility?

```suggestion
    if (typeof this.keyboardDidShowSubscription?.remove === 'function') {
      this.keyboardDidShowSubscription?.remove();
      this.keyboardDidHideSubscription?.remove();
    } else {
      // Keyboard.removeListener()
    }
```

Reply:
There's no backwards compatibility issue as I'm aware, the `remove` function returned by `Keyboard.addListener` is not new - as far as I can tell it's always been there. The change is just that it's now the only way to remove a listener.